pkg/frontend/adminactions/azureactions.go

```diff
@@ -74,6 +74,16 @@ func NewAzureActions(log *logrus.Entry, env env.Interface, oc *api.OpenShiftClus
 
 func (a *azureActions) VMRedeployAndWait(ctx context.Context, vmName string) error {
 	clusterRGName := stringutils.LastTokenByte(a.oc.Properties.ClusterProfile.ResourceGroupID, '/')
+	vm, err := a.virtualMachines.Get(ctx, clusterRGName, vmName, mgmtcompute.InstanceView)
+	if err != nil {
+		return err
+	}
+	vmDisk := vm.StorageProfile.OsDisk
+	if vmDisk.Caching == "ReadOnly" {
+		if vmDisk.DiffDiskSettings.Option == "Local" && vmDisk.DiffDiskSettings.Placement == "CacheDisk" {
```

Comment:
Q: are `DiffDiskSettings` always not nil?

Reply:
I will add the step to check if its `nil`